pulsar-common/src/main/java/org/apache/pulsar/common/util/keystoretls/NettySSLContextAutoRefreshBuilder.java

```diff
@@ -101,8 +100,14 @@ public NettySSLContextAutoRefreshBuilder(String sslProviderString,
         this.tlsAllowInsecureConnection = allowInsecureConnection;
         this.tlsProvider = sslProviderString;
 
-        this.authData = authData;
-
+        if (authData != null) {
+            KeyStoreParams authParams = authData.getTlsKeyStoreParams();
+            if (authParams != null) {
+                keyStoreTypeString = authParams.getKeyStoreType();
+                keyStore = authParams.getKeyStorePath();
+                keyStorePassword = authParams.getKeyStorePassword();
+            }
+        }
         this.tlsKeyStoreType = keyStoreTypeString;
         this.tlsKeyStore = new FileModifiedTimeUpdater(keyStore);
         this.tlsKeyStorePassword = keyStorePassword;
```

Comment:
Nit: is there a reason we're not updating these in the above if block?

Reply:
When the `authData` is available, I override the `keyStoreTypeString`, `keyStore`, and `keyStorePassword` by the `authData.getTlsKeyStoreParams()`.